src/routes/filter/index.ts

```diff
@@ -0,0 +1,30 @@
+import { Request } from 'express';
+import { Health, setRouteHealth } from '../health';
+import { CSEResponse } from '../../server';
+import { IFilterParams } from '../';
+import { applyTracing } from '../../utils/tracer';
+import { IFilterOptions } from '../../models/filter';
+
+export const handler = async (req: Request, res: CSEResponse) => {
+  const { dbClient } = res.locals;
+  const { grades, subject, claimNumber }: IFilterParams = <IFilterParams>req.query;
+  let result: IFilterOptions | undefined;
+  try {
+    await dbClient.connect();
+    if (grades && subject && claimNumber) {
+      result = await dbClient.getTargetShortCodes(grades, subject, claimNumber);
+    } else if (grades && subject) {
+      result = await dbClient.getClaimNumbers(grades, subject);
+    } else {
+      result = await dbClient.getSubjectsAndGrades();
+    }
+    await dbClient.close();
+  } catch (error) {
+    res.status(500);
+    res.send(error);
+  }
+  res.status(200);
+  res.send(result);
```

Comment:
This line and the line above should be moved into the `try/catch` statement, in the `try` branch. If an error is thrown, Express is going to try and set the status and then send the response again.

Reply:
That's a good point. It may take me longer to do this because I'll have to refactor all of the routes and their tests. Another option would be to only send the result at the bottom & set the status code inside the `try/catch`, also letting `result = error` in the `catch`, I prefer a single point of 'return' anyway (even though its not returning anything)